File: CRPL.Data/Applications/InputModels/CopyrightRegistrationInputModel.cs
using CRPL.Contracts.Structs;
using CRPL.Data.Applications.ViewModels;
using CRPL.Data.StructuredOwnership;

namespace CRPL.Data.Applications.InputModels;

public class CopyrightRegistrationInputModel : ApplicationInputModel
{
    public string? Title { get; set; }
    public byte[]? WorkHash { get; set; }
    public string? WorkUri { get; set; }
    public string? Legal { get; set; }
    public List<OwnershipStake>? OwnershipStakes { get; set; }
    public WorkType WorkType { get; set; }
    public int YearsExpire { get; set; }
    public Protections Protections { get; set; }
}

File: CRPL.Web/ClientApp/src/app/_Models/Applications/CopyrightRegistrationInputModel.ts
import {ApplicationInputModel} from "./ApplicationInputModel";
import {OwnershipStake} from "../StructuredOwnership/OwnershipStake";
import {CopyrightType} from "../CopyrightType";
import {WorkType} from "../WorkType";
import {Protections} from "./Protections";


export interface CopyrightRegistrationInputModel extends ApplicationInputModel
{
  Title?: string;
  WorkHash?: string;
  WorkUri?: string;
  Legal?: string;
  OwnershipStakes?: OwnershipStake[];
  YearsExpire?: number;
  CopyrightType?: CopyrightType;
  WorkType?: WorkType;
  Protections: Protections;
}
